packages/arb-validator-core/gobridge/globalInbox.go

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2019-2020, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gobridge
+
+import (
+	"context"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/hashing"
+	"github.com/offchainlabs/arbitrum/packages/arb-validator-core/arbbridge"
+	//"github.com/offchainlabs/arbitrum/packages/arb-validator-core/arbbridge"
+	"github.com/offchainlabs/arbitrum/packages/arb-validator-core/message"
+	"math/big"
+
+	"github.com/offchainlabs/arbitrum/packages/arb-util/common"
+)
+
+type globalInbox struct {
+	inbox   *inbox
+	client  *GoArbAuthClient
+	address common.Address
+}
```

Comment:
Potential race condition in all the `getCurrentBlock()` calls if the block is mined in between the call and the publishing of the event. These two operations need guaranteed atomicity.

Reply:
This race condition still exists since the global inbox is using a different mutex than the mining function.

Reply:
fixed